tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -859,6 +859,27 @@ bool GetHLSLSubobjectKind(clang::QualType type,
   return false;
 }
 
+clang::RecordDecl *GetRecordDeclFromNodeObjectType(clang::QualType ObjectTy) {
+  ObjectTy = ObjectTy.getCanonicalType();
+  DXASSERT(IsHLSLNodeType(ObjectTy), "Expected Node Object type");
+  if (ObjectTy->isStructureOrClassType()) {
+    if (const CXXRecordDecl *CXXRD =
+            ObjectTy.getCanonicalType()->getAsCXXRecordDecl()) {
+
+      if (const ClassTemplateSpecializationDecl *templateDecl =
+              dyn_cast<ClassTemplateSpecializationDecl>(CXXRD)) {
+
+        auto &TemplateArgs = templateDecl->getTemplateArgs();
+        clang::QualType RecType = TemplateArgs[0].getAsType();
+        if (RecordDecl *RD = RecType->getAs<RecordType>()->getDecl()) {
+          return RD;
+        }
```

Comment:
```suggestion
        if (const RecordType *RT = RecType->getAs<RecordType>())
          return RT->getDecl();
```
`getAs<RecordType>()` is the part that may return nullptr if not the right type, so the if should be on that.